src/ape/managers/chain.py

```diff
@@ -431,6 +499,66 @@ def __len__(self) -> int:
 
         return self.provider.get_nonce(self.address)
 
+    @singledispatchmethod
+    def __getitem__(self, index):
+        raise IndexError(f"Can't handle type {type(index)}")
+
+    @__getitem__.register
+    def __getitem_int(self, index: int) -> ReceiptAPI:
+        if index < 0:
+            index += len(self)
+
+        return cast(
+            ReceiptAPI,
+            next(
+                self.query_manager.query(
+                    AccountTransactionQuery(
+                        columns=list(ReceiptAPI.__fields__),
+                        account=self.address,
+                        start_nonce=index,
+                        stop_nonce=index,
+                    )
+                )
+            ),
+        )
+
+    @__getitem__.register
+    def __getitem_slice(self, indices: slice) -> List[ReceiptAPI]:
+        start, stop, step = (
+            indices.start if indices.start else 0,
+            indices.stop if indices.stop else len(self),
+            indices.step if indices.step else 1,
+        )
```

Comment:
Are these able to use `or` since the condition is the value's booleanness?

```suggestion
        start, stop, step = (
            indices.start or 0,
            indices.stop or len(self),
            indices.step or 1,
        )
```